src/lib/native.ts

```diff
@@ -471,21 +471,33 @@ export async function clipboard(
 /**
  * This returns the commandline that was used to start Firefox.
  * You'll get both the binary (not necessarily an absolute path) and flags.
- *
- * On Windows, the main Firefox process is not the direct parent of the native
- * messenger's process, so we need to go up the process tree till we find it.
- * That's rather slow, so this function shouldn't be called unless really necessary.
  */
 export async function ff_cmdline(): Promise<string[]> {
     let output: MessageResp
     if ((await browserBg.runtime.getPlatformInfo()).os === "win") {
-        output = await run(
-            "powershell -Command " +
-                '"$ppid = Get-CimInstance -Property ProcessId,ParentProcessId Win32_Process | Where-Object -Property ProcessId -EQ $PID | Select-Object -ExpandProperty ParentProcessId;' +
-                "$pproc = Get-CimInstance -Property ProcessId,ParentProcessId,Name,CommandLine Win32_Process | Where-Object -Property ProcessId -EQ $ppid;" +
-                "while ($pproc.Name -notmatch 'firefox') { $ppid = $pproc.ParentProcessId;" +
-                '$pproc = Get-CimInstance Win32_Process | Where-Object -Property ProcessId -EQ $ppid}; Write-Output $pproc.CommandLine"',
-        )
+        if (!(await nativegate("0.3.3", false))) {
+            /*
+            If browser binary is not "firefox.exe" but 'browser' is not set,
+            this will hang and use a lot of CPU.
+            See https://github.com/tridactyl/native_messenger/issues/40
+            */
+            output = await run(
+                `powershell -Command \
+"$ppid = Get-CimInstance -Property ProcessId,ParentProcessId Win32_Process | Where-Object -Property ProcessId -EQ $PID | Select-Object -ExpandProperty ParentProcessId;\
+$pproc = Get-CimInstance -Property ProcessId,ParentProcessId,Name,CommandLine Win32_Process | Where-Object -Property ProcessId -EQ $ppid;\
+while ($pproc.Name -notmatch '${await config.get(
```

Comment:
Can't we error out here if no process is called `browser`? That seems like it would fix 95% of cases.

Reply:
Woops, this comment should be one line higher

Reply:
Should be easy enough.